packages/runtime/src/useCases/transport/relationships/CreateRelationship.ts

```diff
@@ -27,18 +29,23 @@ export class CreateRelationshipUseCase extends UseCase<CreateRelationshipRequest
     }
 
     protected async executeInternal(request: CreateRelationshipRequest): Promise<Result<RelationshipDTO>> {
-        const template = await this.relationshipTemplateController.getRelationshipTemplate(CoreId.from(request.templateId));
-        if (!template) {
-            return Result.fail(RuntimeErrors.general.recordNotFound(RelationshipTemplate));
-        }
+        const transformedContent = Serializable.fromUnknown(request.creationContent);
+
+        if (transformedContent instanceof ArbitraryRelationshipCreationContent || transformedContent instanceof RelationshipCreationContent) {
+            const template = await this.relationshipTemplateController.getRelationshipTemplate(CoreId.from(request.templateId));
+            if (!template) {
+                return Result.fail(RuntimeErrors.general.recordNotFound(RelationshipTemplate));
+            }
 
-        const relationship = await this.relationshipsController.sendRelationship({
-            template: template,
-            creationContent: request.creationContent
-        });
+            const relationship = await this.relationshipsController.sendRelationship({ template, creationContent: request.creationContent });
 
-        await this.accountController.syncDatawallet();
+            await this.accountController.syncDatawallet();
+
+            return Result.ok(RelationshipMapper.toRelationshipDTO(relationship));
+        }
 
-        return Result.ok(RelationshipMapper.toRelationshipDTO(relationship));
+        return Result.fail(
+            RuntimeErrors.general.invalidPropertyValue("A relationship creation content must be of type RelationshipCreationContent or ArbitraryRelationshipCreationContent.")
+        );
```

Comment:
Here again think about changing the order of returning an error result and the actual content of the use case.

Reply:
I'd still go for the cleaner if-condition